File: go-client-sdk/pkg/models/shared/codatassessdatacontractsfinancialsmetricsfinancialmetricsdataset.go
package shared

type CodatAssessDataContractsFinancialsMetricsFinancialMetricsDataSet struct {
	Currency   *string                                                        `json:"currency,omitempty"`
	Errors     []CodatAssessDataContractsFinancialsMetricsMetricDataSetError  `json:"errors,omitempty"`
	Metrics    []CodatAssessDataContractsFinancialsMetricsMetric              `json:"metrics,omitempty"`
	PeriodUnit *CodatAssessDataContractsFinancialsMetricsMetricPeriodUnitEnum `json:"periodUnit,omitempty"`
}

File: typescript-client-sdk/src/sdk/models/shared/codatassessdatacontractsfinancialsmetricsfinancialmetricsdataset.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { CodatAssessDataContractsFinancialsMetricsMetricDataSetError } from "./codatassessdatacontractsfinancialsmetricsmetricdataseterror";
import { CodatAssessDataContractsFinancialsMetricsMetric } from "./codatassessdatacontractsfinancialsmetricsmetric";
import { CodatAssessDataContractsFinancialsMetricsMetricPeriodUnitEnum } from "./codatassessdatacontractsfinancialsmetricsmetricperiodunitenum";



export class CodatAssessDataContractsFinancialsMetricsFinancialMetricsDataSet extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=currency" })
  currency?: string;

  @SpeakeasyMetadata({ data: "json, name=errors", elemType: CodatAssessDataContractsFinancialsMetricsMetricDataSetError })
  errors?: CodatAssessDataContractsFinancialsMetricsMetricDataSetError[];

  @SpeakeasyMetadata({ data: "json, name=metrics", elemType: CodatAssessDataContractsFinancialsMetricsMetric })
  metrics?: CodatAssessDataContractsFinancialsMetricsMetric[];

  @SpeakeasyMetadata({ data: "json, name=periodUnit" })
  periodUnit?: CodatAssessDataContractsFinancialsMetricsMetricPeriodUnitEnum;
}
